src/Illuminate/Auth/SessionGuard.php

```diff
@@ -561,7 +561,11 @@ public function logout()
         // listening for anytime a user signs out of this application manually.
         $this->clearUserDataFromStorage();
 
-        if (! is_null($this->user)) {
+        $ableToRefreshToken = ! is_null($this->user) && array_key_exists(
+            $user->getRememberTokenName(), $this->user->getAttributes()
```

Comment:
Partially: https://github.com/laravel/framework/blob/5.2/src/Illuminate/Auth/GenericUser.php#L5

The problem is with `getAttributes()` that not necessarily exists on a GenericUser class, right? In that case I have to make a little change.